docs/wiki/Cryptography/keccak256.md

```diff
@@ -0,0 +1,23 @@
+# Keccak256
+
+Keccak256 is a cryptographic hash function prominently used in the Ethereum blockchain. 
+
+## Brief History
+
+Keccak256 originated as a submission to the [NIST cryptographic hash algorithm competition](https://csrc.nist.gov/projects/hash-functions/nist-hash-function-competition). This competition aimed to identify a new secure hash algorithm to replace SHA-1. The KECCAK algorithm was designed by a team comprising Guido Bertoni, Joan Daemen, Michaël Peeters, and Gilles Van Assche. Their design was notable for its ability to absorb a phrase of arbitrary length and produce a fixed-size output. 
+
+## EVM Implementation
+While SHA-3 is the official version of the algorithm, Ethereum utilizes a modified variant known as Keccak256. This version is tailored for a maximum input of 256 bits. The [Ethereum Virtual Machine (EVM)](https://ethereum.org/en/developers/docs/evm/), a Turing-complete virtual machine, employs Keccak256 for data hashing. 
+
+## Usage in Block Creation and Root Data Structure
+Keccak256 plays a vital role in Ethereum's block creation and in its root data structures. According to the [Ethereum Yellow Paper](https://ethereum.github.io/yellowpaper/paper.pdf), the function is used in several key ways:
+
+- **Block Header Fields**: Various fields in the block header, such as `parentHash` and `stateRoot`, use the Keccak 256-bit hash. This includes hashing the entire header of the parent block, the root node of the state trie, and the root nodes of the trie structures for transactions and receipts.
```

Comment:
Check with @varun-doshi about his work on #28 which might be linked here. These docs should refer to each other but not repeat what was already documented.

Reply:
Can I update it to @varun-doshi section after his section is updated?

Reply:
sure, we can merge it and update later on, I don't want to block you with it